propagators/opentelemetry-propagator-aws-xray/src/AWSXRayPropagator.ts

```diff
@@ -0,0 +1,194 @@
+ * https://https://docs.aws.amazon.com/xray/latest/devguide/xray-concepts.html#xray-concepts-tracingheader>AWS
+ * Tracing header spec</a>
+ *
+ * An example AWS Xray Tracing Header is shown below:
+ * X-Amzn-Trace-Id: Root=1-5759e988-bd862e3fe1be46a994272793;Parent=53995c3f42cd8ad8;Sampled=1
+ */
+export class AWSXRayPropagator implements TextMapPropagator {
+  inject(context: Context, carrier: unknown, setter: TextMapSetter) {
+    const spanContext = getSpanContext(context);
+    if (!spanContext || !isSpanContextValid(spanContext)) return;
+
+    const otTraceId = spanContext.traceId;
+    const timestamp = otTraceId.substring(0, TRACE_ID_FIRST_PART_LENGTH);
+    const randomNumber = otTraceId.substring(TRACE_ID_FIRST_PART_LENGTH);
+
+    const parentId = spanContext.spanId;
+    const samplingFlag =
+      (TraceFlags.SAMPLED & spanContext.traceFlags) === TraceFlags.SAMPLED
+        ? IS_SAMPLED
+        : NOT_SAMPLED;
+    // TODO: Add OT trace state to the X-Ray trace header
+
+    const traceHeader = `Root=1-${timestamp}-${randomNumber};Parent=${parentId};Sampled=${samplingFlag}`;
+    setter.set(carrier, AWSXRAY_TRACE_ID_HEADER, traceHeader);
+  }
+
+  extract(context: Context, carrier: unknown, getter: TextMapGetter): Context {
+    const spanContext = this.getSpanContextFromHeader(carrier, getter);
+    if (!isSpanContextValid(spanContext)) return context;
+
+    return setSpanContext(context, spanContext);
+  }
+
+  fields(): string[] {
+    return [AWSXRAY_TRACE_ID_HEADER];
+  }
+
+  private getSpanContextFromHeader(
+    carrier: unknown,
+    getter: TextMapGetter
+  ): SpanContext {
+    const traceHeader = getter.get(carrier, AWSXRAY_TRACE_ID_HEADER);
+    if (!traceHeader || typeof traceHeader !== 'string')
+      return INVALID_SPAN_CONTEXT;
+
+    let pos = 0;
+    let trimmedPart: string;
+    let parsedTraceId = INVALID_TRACEID;
+    let parsedSpanId = INVALID_SPANID;
+    let parsedTraceFlags = null;
+    while (pos < traceHeader.length) {
+      const delimiterIndex = traceHeader.indexOf(TRACE_HEADER_DELIMITER, pos);
+      if (delimiterIndex >= 0) {
+        trimmedPart = traceHeader.substring(pos, delimiterIndex).trim();
+        pos = delimiterIndex + 1;
+      } else {
+        //last part
+        trimmedPart = traceHeader.substring(pos).trim();
+        pos = traceHeader.length;
+      }
+      const equalsIndex = trimmedPart.indexOf(KV_DELIMITER);
+
+      const value = trimmedPart.substring(equalsIndex + 1);
+
+      if (trimmedPart.startsWith(TRACE_ID_KEY)) {
+        parsedTraceId = AWSXRayPropagator._parseTraceId(value);
+      }
+      if (trimmedPart.startsWith(PARENT_ID_KEY)) {
+        parsedSpanId = AWSXRayPropagator._parseSpanId(value);
+      }
+      if (trimmedPart.startsWith(SAMPLED_FLAG_KEY)) {
+        parsedTraceFlags = AWSXRayPropagator._parseTraceFlag(value);
+      }
+    }
+    if (parsedTraceFlags === null) {
+      return INVALID_SPAN_CONTEXT;
```

Comment:
According to readme the sampled part is optional but it seems it is required here.

Reply:
I have removed the text about it, since with OpenTelemetry all parts are required, we basically require them here (or start a new trace otherwise)